pkg/venafi/tpp/connector.go

```diff
@@ -1231,31 +1265,53 @@ func (c *Connector) getIdentity(userName string) (*policy.IdentityEntry, error)
 		return nil, err
 	}
 
-	return c.getIdentityMatching(resp.Identities, userName)
-}
+	// When TPP looks for a username that matches the filter, an implicit
+	// wildcard is added to the end of the filter string. For example, imagining
+	// that `jsmith` and `jsmithson` are existing identities, searching for
+	// `jsmith` will return both `jsmith` and `jsmithson`. In the case of local
+	// identities, `jsmith` will always be returned first. But in the case of AD
+	// and LDAP, the order of these results may be different, and `jsmithson`
+	// may be unexpectedly returned first. This same problem may appear when an
+	// AD or LDAP provider has been configured to access the local identities:
+	// `jsmithson` may get returned first if `jsmithson` only exists in AD
+	// (because the AD results are returned before the local identities).
+	//
+	// The wildcard problem only affects usernames, not emails. That's because
+	// the LDAP query recommended for enabling user search by email in the
+	// Venafi Configuration Console is based on exact match, unlike `anr` used
+	// for searching usernames. Thus, we do not need to check for an exact match
+	// when an email is provided.
 
-func (c *Connector) getIdentityMatching(identities []policy.IdentityEntry, identityName string) (*policy.IdentityEntry, error) {
-	var identityEntryMatching *policy.IdentityEntry
+	// When the filter isn't an email, we consider that it is a username.
+	_, err = mail.ParseAddress(filter)
+	isEmail := err == nil
+	isUsername := !isEmail
```

Comment:
It's not required to have two flags, only with one is enough given the origin of these values is the result of `mail.ParseAdress()` function so the flag that should keep is `isEmail` and use the resulting value to do the respective logic if the filter was an email or not

Reply:
I agree, I'll revert back to just `isEmail`.